artemis-server/src/main/java/org/apache/activemq/artemis/spi/core/security/ActiveMQJAASSecurityManager.java

```diff
@@ -141,8 +141,9 @@ private Subject getAuthenticatedSubject(final String user,
          try {
             lc.login();
          } catch (LoginException e) {
-            Throwable rootCause = ExceptionUtil.getRootCause(e);
+            Throwable rootCause = ExceptionUtils.getRootCause(e);
             if (rootCause instanceof NoCacheLoginException) {
+               logger.debug("Handling LoginException with NoCacheLoginException root cause", e);
```

Comment:
Do you think inspecting a `String` is preferable to an `instanceof`? Perhaps that would be more intuitive.

Reply:
@jbertram thanks for pointing me to the reason of the current implementation. If I understand correctly JAAS converts NoCacheLoginException into LoginException because NoCacheLoginException is not a subclass of LoginException.

If NoCacheLoginException would inherit from LoginException and ActiveMQSecurityManager5.authenticate would throw LoginException we could avoid to set and check the root cause, we could just throw and catch NoCacheLoginException. Is that right?

Reply:
We don't need a new interface because class implementing an interface don't have to throw exceptions defined in the interface.

Reply:
I just pushed with this update.